File: services/saas-api/src/saas_api/schemas/telegram.py
from pydantic import BaseModel, ConfigDict, Field


class EndUserSummary(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    tenant_id: str = Field(alias="tenantId")
    telegram_id: str = Field(alias="telegramId")
    telegram_username: str | None = Field(default=None, alias="telegramUsername")
    first_name: str | None = Field(default=None, alias="firstName")
    last_name: str | None = Field(default=None, alias="lastName")
    language_code: str | None = Field(default=None, alias="languageCode")


class ValidateInitDataRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    tenant_slug: str = Field(alias="tenantSlug")
    init_data: str = Field(alias="initData")


class ValidateInitDataResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    user: EndUserSummary

File: packages/api-contracts/src/telegram.ts
import { z } from "zod";

export interface EndUserSummary {
  id: string;
  tenantId: string;
  telegramId: string;
  telegramUsername?: string | null;
  firstName?: string | null;
  lastName?: string | null;
  languageCode?: string | null;
}

export interface ValidateInitDataRequest {
  tenantSlug: string;
  initData: string;
}

export interface ValidateInitDataResponse {
  user: EndUserSummary;
}

export const endUserSummarySchema = z.object({
  id: z.string().min(1),
  tenantId: z.string().min(1),
  telegramId: z.string().min(1),
  telegramUsername: z.string().nullable().optional(),
  firstName: z.string().nullable().optional(),
  lastName: z.string().nullable().optional(),
  languageCode: z.string().nullable().optional(),
});

export const validateInitDataResponseSchema = z.object({
  user: endUserSummarySchema,
});
